src/encoding/binarydata.ts

```diff
@@ -53,3 +64,22 @@ export function bytesToHex(byteArray: Uint8Array): string {
     .map((i) => i.toString(16).padStart(2, '0'))
     .join('');
 }
+
+/**
+ * Convert a hex string to Uint8Array for Node.js and browser environments.
+ * @returns A Uint8Array
+ */
+export function hexToBytes(hexString: string): Uint8Array {
+  if (isNode()) {
+    return Buffer.from(hexString, 'hex');
+  }
+  let hex = hexString;
+  if (hexString.length % 2 !== 0) {
+    hex = hexString.padStart(1, '0');
+  }
+  const byteArray = new Uint8Array(hex.length / 2);
+  for (let i = 0, j = 0; i < hex.length / 2; i++, j += 2) {
+    byteArray[i] = parseInt(hex.slice(j, j + 2), 16);
```

Comment:
```suggestion
  for (let i = 0; i < hex.length / 2; i++) {
    byteArray[i] = parseInt(hex.slice(2*i, 2*i + 2), 16);
```

Reply:
Either way seems fine--this way you don't need double iter vars though which feels cleaner to me.